fonts: Add synthetic bold face support for Windows

minghuaw opened the pull request:
This PR follows #39519 and adds synthetic bold face support for Windows platform.

Testing: There are existing WPT testcases for font synthesis (wpt/css/css-fonts/font-synthesis-*). A new test that checks for "double emboldening" (a bold font gets emboldened again) is getting added to the upstream WPT repo (https://github.com/web-platform-tests/wpt/pull/55313)
Depends on: https://github.com/servo/dwrote-rs/pull/70, https://github.com/servo/dwrote-rs/pull/71
Part of #39637

minghuaw committed:
impl synthetic bold face for windows

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
do not set synthetic_bold if font supports variations

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
inline conversion to DWRITE_FONT_SIMULATIONS flag

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
use git deps temporarily

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw commented on Cargo.toml:
```diff
@@ -232,6 +232,8 @@ codegen-units = 1
 # markup5ever = { path = "../html5ever/markup5ever" }
 # web_atoms = { path = "../html5ever/web_atoms" }
 # xml5ever = { path = "../html5ever/xml5ever" }
+dwrote = { git = "https://github.com/minghuaw/dwrote-rs" } # TODO: remove once dwrote-rs/#70 lands
```
Reminder: remove once https://github.com/servo/dwrote-rs/pull/70 lands

xiaochengh reviewed (approved):
LGTM with nits

Commit:
remove tmp git dep and address comments

Signed-off-by: minghuaw <wuminghua7@huawei.com>

jdm commented on Cargo.toml:
```diff
@@ -232,6 +232,7 @@ codegen-units = 1
 # markup5ever = { path = "../html5ever/markup5ever" }
 # web_atoms = { path = "../html5ever/web_atoms" }
 # xml5ever = { path = "../html5ever/xml5ever" }
+dwrote = { git = "https://github.com/servo/dwrote-rs" } # TODO: remove once new version gets published
```
0.11.5 is published.

Commit:
remove temporary git dep

Signed-off-by: minghuaw <wuminghua7@huawei.com>

Commit:
update Cargo.lock

Signed-off-by: minghuaw <wuminghua7@huawei.com>

Commit:
add flags to FontInstance cache keys

Signed-off-by: minghuaw <wuminghua7@huawei.com>

minghuaw committed:
check if font is already bold in PlatformFont

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
add test checking for double emboldening

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
update expectation for linux

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
update manifest & fix test-tidy

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
make apply synthetic bold logic easier to read

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

Commit:
make synthetic_bold easier to read

Signed-off-by: minghuaw <wuminghua7@huawei.com>

minghuaw committed:
Merge branch 'main' into feat/synth-bold-windows

minghuaw commented:
> dwrote was updated in #39699.

Thanks! Updated the branch to main. This PR is ready for another quick review @xiaochengh (I've added `FontInstanceFlags` as part of the font instance map key since the last review).

minghuaw committed:
rename to entry_key

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
update WPT expectations

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
Merge branch 'main' into feat/synth-bold-windows

minghuaw committed:
fmt

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>

minghuaw committed:
remove font_synthesis_weight_static_bold test

Signed-off-by: Minghua Wu <michael.wu1107@gmail.com>
